### ISSUE TITLE
Are client credentials required for `GoogleDrive.login_with_oauth`?

### ISSUE THREAD

**Issue by joeltaylor** [gimite/google-drive-ruby#243]
I'm currently using `omniauth-google-oauth2` along with `devise` to retrieve and store a user's access token in my application.
 
From what I've gathered, given the correct scope, I should be able to pass the `access_token` to `GoogleDrive.login_with_oauth` to make authenticated requests on behalf of the user? 

When I attempt this I end up getting the following error:

```
Caught error Authorization failed.  Server message:
{
 "error": "unsupported_grant_type",
 "error_description": "Invalid grant_type: "
}
```

So far, I've had to do something like this in order to use the already retrieved access_token.
```
    client_credentials = Google::Auth::UserRefreshCredentials.new(
      client_id: CLIENT_ID
      client_secret: CLIENT_SECRET,
      scope: [ "https://www.googleapis.com/auth/drive" ],
      access_token: @credentials["token"],
      refresh_token: @credentials["refresh_token"] 
    )

    session = GoogleDrive::Session.from_credentials(client_credentials)
   
    # Downloads to a local file.
    ...
```

**Comment by gimite** [comment]
Yeah that should work. What exactly is the code you wrote with login_with_oauth?

**Comment by joeltaylor** [comment]
I  tried a few variations:
```
require "google_drive"

# Other variations I tried
# session = GoogleDrive::Session.from_access_token(@credentials["token"])
# session = GoogleDrive.login_with_oauth(@credentials["token"])
session = GoogleDrive::Session.login_with_oauth(@credentials["token"])


# Error is thrown when trying to make any calls with the session.
session.files.each do |file|
  p file.title
end
```

I may be making the incorrect assumption that I don't need to define the client credentials and scope since I already have the token.  So it might be because the request options aren't set? 

For example, with a session generated from any of the variations above I get these request options (`session.drive.request_options`).
```
--- |
  --- !ruby/struct:Google::Apis::RequestOptions
  authorization: !ruby/object:Google::Auth::UserRefreshCredentials
    authorization_uri: !ruby/object:Addressable::URI
      validation_deferred: false
      scheme: https
      host: accounts.google.com
      path: "/o/oauth2/auth"
      authority: accounts.google.com
      uri_string: https://accounts.google.com/o/oauth2/auth
    token_credential_uri: !ruby/object:Addressable::URI
      validation_deferred: false
      scheme: https
      host: www.googleapis.com
      path: "/oauth2/v3/token"
      authority: www.googleapis.com
      uri_string: https://www.googleapis.com/oauth2/v3/token
    client_id: 
    client_secret: 
    code: 
    expires_at: 
    expires_in: 
    issued_at: 
    issuer: 
    password: 
    principal: 
    redirect_uri: 
    scope: 
    state: 
    username: 
    expiry: 60
    extension_parameters: {}
    additional_parameters: {}
    access_token: REMOVED_TOKEN
    grant_type: 
    refresh_token: 
  retries: 0
  header: 
  timeout_sec: 100000000
  open_timeout_sec: 100000000
```

**Comment by gimite** [comment]
Thanks, I found that it is broken with the new version of googleauth gem. Made a fix and pushed google_drive 2.1.2 with the fix.
